WD-Firmware/psp/api_inc/psp_interrupt_api.h

```diff
@@ -0,0 +1,112 @@
+/*
+* SPDX-License-Identifier: Apache-2.0
+* Copyright 2019 Western Digital Corporation or its affiliates.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http:*www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+* @file   psp_interrupt_api.h
+* @author Ronen Haen
+* @date   20.05.2019
+* @brief  The file defines the psp interrupt interfaces
+*/
+#ifndef __PSP_INTERRUPT_API_H__
+#define __PSP_INTERRUPT_API_H__
+
+/**
+* include files
+*/
+#include "common_types.h"
+
+/**
+* definitions
+*/
+
+/**
+* macros
+*/
+
+/**
+* types
+*/
+
+/* */
+typedef enum pspInterruptCause
+{
+   E_USER_SOFTWARE_CAUSE       = 0,
+   E_SUPERVISOR_SOFTWARE_CAUSE = 1,
+   E_RESERVED_SOFTWARE_CAUSE   = 2,
+   E_MACHINE_SOFTWARE_CAUSE    = 3,
+   E_USER_TIMER_CAUSE          = 4,
+   E_SUPERVISOR_TIMER_CAUSE    = 5,
+   E_RESERVED_TIMER_CAUSE      = 6,
+   E_MACHINE_TIMER_CAUSE       = 7,
+   E_USER_EXTERNAL_CAUSE       = 8,
+   E_SUPERVISOR_EXTERNAL_CAUSE = 9,
+   E_RESERVED_EXTERNAL_CAUSE   = 10,
+   E_MACHINE_EXTERNAL_CAUSE    = 11,
+   E_LAST_CAUSE
+} pspInterruptCause_t;
+
+/* Exceptions */
+typedef enum pspExceptionCause
+{
+	E_EXC_INSTRUCTION_ADDRESS_MISALIGNED           = 0,
```

Comment:
not so important ... but spaces (we should have 2 spaces for tabs, and not using tabs)